### PR TITLE
ENH: improve failure messages and behavior on xspress3 unstage

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by danielballan** [comment]
The fix looks right, but shouldn't this PR go into `nslsii`, here: https://github.com/NSLS-II/nslsii/blob/master/nslsii/detectors/xspress3.py#L126-L143?

File: hxntools/detectors/xspress3.py [hunk 1]
```diff
@@ -116,17 +116,24 @@ def unstage(self):
...
+            # correctly and it is expecting to capture more points than it
+            # was triggered to take.
             while self.capture.get() == 1:
                 i += 1
                 if (i % 50) == 0:
                     logger.warning('Still capturing data .... waiting.')
                 time.sleep(0.1)
                 if i > 150:
                     logger.warning('Still capturing data .... giving up.')
+                    logger.warning('Check that the xspress3 is configure to take the right.'
```
Comment (mrakitin) [inline hxntools/detectors/xspress3.py]:
```suggestion
                    logger.warning('Check that the xspress3 is configured to take the right '
```

File: hxntools/detectors/xspress3.py [hunk 2]
```diff
@@ -116,17 +116,24 @@ def unstage(self):
...
+                                   'number of frames '
+                                   f'(it is trynig to take {self.parent.settings.num_images.get()})')
```
Comment (mrakitin) [inline hxntools/detectors/xspress3.py]:
```suggestion
                                   f'(it is trying to take {self.parent.settings.num_images.get()})')
```

**Review by mrakitin — COMMENTED** [review]
Looks good to me modulo the typos I suggested to fix below. I think it can go in together with #18 in the next release.

**Pull request merged**